### DISCUSSION TITLE
Lesson12: Handler Setup failed.

### DISCUSSION THREAD

**Discussion by RjtSinghal** [Cyfrin/foundry-full-course-cu#1626] [General]
This is my code for setup function when I try to link handler but I am getting error. Below is my setup function code. I have checked the handler constructor is correct. I am not able to fisure out the error,
```
function setUp() external {
        deployer = new DeployDSC();
        (dsc, dsce, config) = deployer.run();
        (,, weth, wbtc,) = config.activeNetworkConfig();
        // targetContract(address(dsce));
        handler = new Handler(dsce, dsc);
        targetContract(address(handler));
        // only call redeemCollateral, when there is a collateral to redeem.
    }
```

ERROR:
Failing tests:
Encountered 1 failing test in test/fuzz/Invariants.t.sol:Invariants
[FAIL. Reason: setup failed: EvmError: Revert] setUp() (gas: 0)

**Comment by EngrPips** [comment]
Hello @RjtSinghal, please run the test again but now append `-vvvv` to the end of the test so it shows an output with a stack trace of what happened. Like that, we might be able to figure out what went wrong quickly.

**Reply by RjtSinghal** [reply]
Here is the console image after running tests using -vvvv.

<img width="1907" alt="Screenshot 2024-04-21 at 2 56 59 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/22500665/be4d5ff9-8f2c-440c-8f4b-61f0fb4b2cf5">

The test is failing for targetContract(address(handler)) only. It is working for targetcontract(address(dsce)).

**Comment by RjtSinghal** [comment]
resolved

**Reply by EngrPips** [reply]
Excellent, I am glad you figured it out. Keep crushing it, friend
